File: perl/19_bit_positions.pl
#!/usr/bin/env perl

use strict;
use warnings;

sub bit {
	my $n = shift @_;
	my $p = shift @_;

	$p--;

	return ( $n & 1 << $p ) >> $p;
}

open my $fh, '<', $ARGV[0];

for my $line ( <$fh> ) {
	chomp $line;
	next if $line eq '';

	my @parts = split /,/, $line;
	my $n     = $parts[0];
	my $p1    = $parts[1];
	my $p2    = $parts[2];

	print bit( $n, $p1 ) eq bit( $n, $p2 ) ? "true\n" : "false\n";
}

close $fh;

File: python/19_bit_positions.py
import sys

def bit(n, p):
    return (n & (1 << (p - 1))) >> (p - 1)

with open(sys.argv[1], 'r') as fh:
    for line in fh.readlines():
        line = line.strip()
        if line == '':
            continue

        (n, p1, p2) = [int(v) for v in line.split(',')]

        print 'true' if bit(n, p1) == bit(n, p2) else 'false'

